src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h

```diff
@@ -1235,6 +1236,46 @@ typedef struct {
                      //!< RTCDataChannel object with the same id as the other peer.
 } RtcDataChannelInit, *PRtcDataChannelInit;
 
+
+/////////////////////////////////////////////////////
+/// Metrics/Stats Related structures
+/////////////////////////////////////////////////////
+
+/**
+ * @brief Collection of ICE related stats
+ */
+typedef struct {
+    PRTCIceServerStats pIceServerStats; //!< Server related stats. Reference in Stats.h
+    PRTCIceCandidateStats pIceCandidateStats; //!< Single candidate stats. Reference in Stats.h
+    PRTCIceCandidatePairStats pIceCandidatePairStats; //!< Candidate pair stats. Reference in Stats.h
+} IceStats, *PIceStats;
+
+/**
+ * @brief Collection of RTP stream related stats
+ */
+typedef struct {
+    PRTCRemoteInboundRtpStreamStats pInboundStats; //!< Inbound RTP Stats. Reference in Stats.h
+    PRTCOutboundRtpStreamStats pOutboundStats; //!< Outbound RTP Stats. Reference in Stats.h
+    PRTCTransportStats pTransportStats; //!< Transport stats. Reference in Stats.h
+} StreamStats, *PStreamStats;
+
+/**
+ * @brief SignalingStats Collection of signaling related stats. Can be expanded in the future
+ */
+typedef struct {
+    SignalingClientStats signalingClientMetrics; //!< Signaling client metrics stats. Reference in Stats.h
```

Comment:
Not sure if we need two levels here - perhaps flat signaling stats might be enough

Reply:
Yes it is. Just having it in 2 levels incase we have any new signaling related structure in the future. That is the reason I mentioned can be expanded in the future

Reply:
I don't think this indirection really is necessary even for the future expansion. The version will provide means to expand and still support backward compat.